### PR TITLE
Add CI configuration to run tests with Python 3 interpreter

### LINKED ISSUE #3872: CI: bazel-tests should add a Python 3.x configuration
The current [bazel-tests CI job](https://github.com/bazelbuild/bazel/blob/master/scripts/ci/bazel-tests.json) does not run against a Python 3.x interpreter, so issues like #3816 happen.  We should add a configuration that does so, via the `--python_path` flag now, and perhaps via the `--python_top` flag later.

An alternative would be to add an ArchLinux configuration, which has `/usr/bin/python` point to a Python 3.x interpreter.

### PR DESCRIPTION
**Pull request by duggelz**
Fixes #3872

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by duggelz** [comment]
@damienmg @mattmoor How do I poke the CI system to test this?

**Comment by mattmoor** [comment]
test this please

**Comment by mattmoor** [comment]
I guess I lack the magic powers on this repo.

**Comment by damienmg** [comment]
@duggelz : you need to make the google org public on your github account to have the right to trigger a build. And then "test this please" is the magic sentence (it also auto build at each new push if the pr comes from you).

**Comment by duggelz** [comment]
Huh, I had no idea they were private.  Fixed.  I wonder how much past confusion this has caused :(

test this please

**Comment by duggelz** [comment]
test this please

Fiddling with permissions, currently getting this error:

```
Caused by: com.google.api.client.http.HttpResponseException: 400 Bad Request
{
  "error" : "invalid_grant",
  "error_description" : "Invalid JWT Signature."
}
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1070)
	at com.google.auth.oauth2.ServiceAccountCredentials.refreshAccessToken(ServiceAccountCredentials.java:317)
```

**Comment by damienmg** [comment]
Were do you get that error from?

**Comment by duggelz** [comment]
Permission error is gone, now I'm seeing the Python 3 errors I expect to see:

http://ci.bazel.io/blue/rest/organizations/jenkins/pipelines/PR/pipelines/bazel-tests/runs/121/nodes/26/log/?start=0

Along with a FreeBSD error that seems completely unrelated:

http://ci.bazel.io/blue/rest/organizations/jenkins/pipelines/PR/pipelines/bazel-tests/runs/121/nodes/28/log/?start=0

Could you look at the config and Jenkins output and confirm that I am indeed running the tests under both Python 2 and Python 3?  I.e. once without `--python_path` and once with `--python_path`?  Because I look at the pretty graphic and I only see one `node=linux-x86_64` under "Run Configurations", but I would sort of expect to see two.

File: scripts/ci/bazel-tests.json [hunk 1]
```diff
@@ -4,6 +4,12 @@
     {
         "configurations": [
             {"node": "linux-x86_64"},
+            {
+             "node": "linux-x86_64",
```
Comment (damienmg) [inline scripts/ci/bazel-tests.json]: add "python_version": "3"

File: scripts/ci/bazel-tests.json [hunk 2]
```diff
@@ -4,6 +4,12 @@
     {
         "configurations": [
             {"node": "linux-x86_64"},
```
Comment (damienmg) [inline scripts/ci/bazel-tests.json]: add "python_version": "2"

File: scripts/ci/bazel-tests.json [hunk 3]
```diff
@@ -4,6 +4,12 @@
...
+            {
+             "node": "linux-x86_64",
+             "parameters": {
+                 "build_opts": ["--python_path=/usr/bin/python3"]
+              }
+            },
```
Comment (damienmg) [inline scripts/ci/bazel-tests.json]: I recommend removing the non-linux configuration so you get a less noisy result. E.g. freebsd is a breakage at HEAD.

File: scripts/ci/bazel-tests.json (hunk 2 above)
Comment (duggelz) [reply scripts/ci/bazel-tests.json]: Added

File: scripts/ci/bazel-tests.json (hunk 1 above)
Comment (duggelz) [reply scripts/ci/bazel-tests.json]: Added

File: scripts/ci/bazel-tests.json (hunk 3 above)
Comment (duggelz) [reply scripts/ci/bazel-tests.json]: Done

**Comment by duggelz** [comment]
Can we submit this with failing tests, or do we have to fix all the Python 3 breakages first, then submit this?

File: scripts/ci/bazel-tests.json [hunk 4]
```diff
@@ -3,7 +3,18 @@
 [
     {
         "configurations": [
-            {"node": "linux-x86_64"},
+            {
+             "node": "linux-x86_64",
+             "parameters": {
+                 "python_version": "2",
+             },
+            {
```
Comment (duggelz) [inline scripts/ci/bazel-tests.json]: I'm not sure what indentation it wants here.

**Comment by duggelz** [comment]
test this please

(For Bazel CI system)

**Comment by duggelz** [comment]
Tests now pass, except for the FreeBSD node which is unrelatedly broken at HEAD, from what I can tell.

**Pull request closed without merging**